prov/verbs/src/verbs_info.c

```diff
@@ -1045,38 +1110,252 @@ static int fi_ibv_get_matching_info(uint32_t version, const char *dev_name,
+		if (ret) {
+			VERBS_INFO(FI_LOG_CORE,
+				   "ofi_ns_init returns %d\n", ret);
+			ret = -FI_ENODATA;
+			goto fn1;
+		}
+
+		if (service)
+			svc = atoi(service);
+		svc0 = svc;
+		dest_addr = (struct ofi_ib_ud_ep_name *)
+			ofi_ns_resolve_name(&ns, node, &svc);
+		if (dest_addr) {
+			VERBS_INFO(FI_LOG_CORE,
+				   "'%s:%u' resolved to <gid <interface_id=%x, "
+				   "subnet_prefix=%x>, lid=%d>:%d\n",
+				   node, svc0, dest_addr->gid.global.interface_id,
+				   dest_addr->gid.global.subnet_prefix,
+				   dest_addr->lid, svc);
+		} else {
+			VERBS_INFO(FI_LOG_CORE,
+				   "failed to resolve '%s:%u'.\n", node, svc);
+			ret = -FI_ENODATA;
+			/* Here possible that `src_addr` isn't a NULL */
+			goto fn2;
+		}
+	}
+
+	for (iter_info = *info; iter_info; iter_info = iter_info->next) {
+		if (iter_info->ep_attr->type != FI_EP_DGRAM)
+			continue;
+		iter_info->addr_format = fmt;
+		if (src_addr) {
+			if (fmt == FI_ADDR_STR) {
+				size_t len = 0;
+				iter_info->src_addr =
+					fi_ibv_dgram_ep_name_to_string(src_addr,
+								       &len);
+				if (!iter_info->src_addr) {
+					ret = -FI_ENOMEM;
+					goto fn2;
+				}
+				iter_info->src_addrlen = len;
+			} else {
+				iter_info->src_addr = calloc(1, sizeof(*src_addr));
+				if (!iter_info->src_addr) {
+					ret = -FI_ENOMEM;
+					goto fn2;
+				}
+				memcpy(iter_info->src_addr, src_addr,
+				       sizeof(*src_addr));
+				iter_info->src_addrlen = sizeof(*src_addr);
+			}
+		}
+		if (dest_addr) {
+			if (fmt == FI_ADDR_STR) {
+				size_t len = 0;
+				iter_info->dest_addr =
+					fi_ibv_dgram_ep_name_to_string(dest_addr,
+								       &len);
+				if (!iter_info->dest_addr) {
+					ret = -FI_ENOMEM;
+					goto fn2;
+				}
+				iter_info->dest_addrlen = len;
+			} else {
+				iter_info->dest_addr = calloc(1, sizeof(*src_addr));
+				if (!iter_info->dest_addr) {
+					ret = -FI_ENOMEM;
+					goto fn2;
+				}
+				memcpy(iter_info->dest_addr, dest_addr, sizeof(*dest_addr));
+				iter_info->dest_addrlen = sizeof(*dest_addr);
+			}
+		}
+	}
+
+	/* `fi_info::src_addr` and `fi_info::dest_addr` is freed
+	 * in the `fi_freeinfo` function */
+fn2:
+	if (src_addr)
+		free(src_addr);
+	if (dest_addr)
+		free(dest_addr);
+fn1:
+	return ret;
+}
+
+static int fi_ibv_get_match_infos_on_device_name(uint32_t version, const char *node,
+						 const char *service, uint64_t flags,
+						 const struct fi_info *hints,
+						 const struct fi_info **raw_info,
+						 struct fi_info **info)
 {
 	struct rdma_cm_id *id = NULL;
 	struct rdma_addrinfo *rai;
-	const struct fi_info *raw_info = fi_ibv_util_prov.info;
+	int ret, sock_addr_handled = 0, ib_ud_addr_handled = 0;
 	const char *dev_name = NULL;
-	int ret;
 
-	ret = fi_ibv_create_ep(node, service, flags, hints, &rai, &id);
-	if (ret)
-		goto out;
+	/* This is case when only IB UD addresses are passed */
+	if (hints && (FI_IBV_IS_IB_UD_ADDR(hints->dest_addr, hints->dest_addrlen) ||
+	    FI_IBV_IS_IB_UD_ADDR(hints->src_addr, hints->src_addrlen))) {
```

Comment:
The addr_format field should indicate if this is an IB UD address format.  Is there plans to support mapping a UDP-IP (sockaddr) address to IB UD?  If so, additional checking may be necessary.

Reply:
> Is there plans to support mapping a UDP-IP (sockaddr) address to IB UD?

No, there are no plans to support mapping just now.

Thanks for this comment :)